### PR TITLE
Add cron jobs

### PR DESCRIPTION
**Pull request by jagonalez**
Adds a cron plugin
- cron jobs can be added via toml configuration:
```
[[ cron ]]
  schedule = "* * * * *" # 5 or 6 fields (6 fields is quartz specification for seconds)
  [cron.job]
    type = "Scan" # required
 
```
- plugin must be enabled in order to run:
```
[cron_plugin]
enabled = true
```
- only supports adding cronjobs when starting faktory, modifying jobs requires restarting

### CODE REVIEW

File: cron/cron.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+		return fmt.Errorf("Start: cannot add cron jobs: %v", err)
+	}
+
+	util.Infof("Started cron plugin, registered %d jobs", len(c.Options.CronJobs))
+	go func() {
+		// wait for signal and stop cron
+		<-c.Server.Stopper()
+		c.Cron.Stop()
+	}()
+	return nil
```
Comment (microsoftly) [inline cron/cron.go:65]: [nit] I tend to prefer to have `return` with a newline unless it's the online line in a block

File: cron/cron.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+	if err != nil {
+		return cronJob, fmt.Errorf("parseCronJob: unable to marshal job: %v", err)
+	}
+
+	var job client.Job
+
+	// validate jobData is correct
+	err = json.Unmarshal(jobData, &job)
+
+	if err != nil {
```
Comment (microsoftly) [inline cron/cron.go:203]: [nit] no newline between err value assignment and nil check

File: cron/cron_test.go [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+	enabledConfig = `
+	[cron_plugin]
+	  enabled = true
+	[[ cron ]]
+	  schedule = "* * * * *" # every minute
+	  [cron.job]
+	    type = "minutely_test_job"
+		retry = 5
+		queue = "test"
+		args = [1, 3]
```
Comment (microsoftly) [inline cron/cron_test.go:24]: I'd like to see assertions for these args
I'd also like to see args that are dictionaries, eg
```toml
    [[cron.job.args]]
      hello = "world"
    [[cron.job.args]]
      hello = "human"
```

**Review by microsoftly — APPROVED** [review]
This works for me
Screen Shot 2022-01-04 at 12 21 32 PM

I can get the payload to have args when adding the following
```toml
[[ cron ]]
schedule = "* * * * *"
  [cron.job]
    type = "JobType"
    queue = "queue"
    retry = 3
    [[cron.job.args]]
      hello = "world"
    [[cron.job.args]]
      hello = "human"
    [cron.job.custom]
      value = true
```
ex
Screen Shot 2022-01-04 at 12 47 29 PM

I think it's worth adding those docs into the readme.

I'm approving for now to unblock you - I trust your test writing that you don't need me to recheck. Please ping me if you would like another round of eyes.

# tl;dr
- [ ] add how to set args to docs
- [ ] add tests asserting against args as an array of primitives
- [ ] add tests asserting against args as an array of dictionary values
